crates/sui-node/src/lib.rs

```diff
@@ -643,7 +643,7 @@ impl SuiNode {
             None
         };
 
-        let chain_identifier = ChainIdentifier::from(*genesis.checkpoint().digest());
+        let chain_identifier = epoch_store.get_chain_identifier();
```

Comment:
the new and old code would always have the same value, right?

Reply:
Ok I see what you mean. I think it's a bit confusing that the `get_chain_identifier` on the epoch store it returns the original chain , and the `get_chain` provides the overwrite. So IIUC the chain gets overwritten within the epoch store, but not on the rest of the code base? Is there a reason for this @mystenmark ?

Reply:
yes, the new pattern is that we distinguish between the chain id (a digest) and the nominal chain (mainnet, testnet, etc). The id can never change, as that would involve lying about the digest of the genesis checkpoint. The nominal chain can be overridden. So I still think there is no behavior change caused by this code, but its definitely a good cleanup anyway.

Reply:
Makes sense. Ok leaving like this then for now.